### PR TITLE
fix(edit-article): check old data before update revision count

### PR DESCRIPTION
**Pull request by gary02**

### CODE REVIEW

File: src/mutations/article/editArticle.ts [hunk 1]
```diff
@@ -159,7 +159,10 @@ const resolver: GQLMutationResolvers['editArticle'] = async (
   /**
    * Tags
    */
-  if (tags !== undefined) {
+  if (
+    tags !== undefined &&
+    (tags ?? []).toString() !== articleVersion.tags.toString()
```
Comment (robertu7) [inline src/mutations/article/editArticle.ts:164]: `.sort()` to ensure a stable array?

File: src/mutations/article/editArticle.ts [hunk 2]
```diff
@@ -168,38 +171,45 @@ const resolver: GQLMutationResolvers['editArticle'] = async (
...
    * Connection
    */
   if (collection !== undefined) {
-    checkRevisionCount(article.revisionCount + 1)
-    updateRevisionCount = true
+    const connections = (collection ?? []).map(
+      (globalId) => fromGlobalId(globalId as unknown as string).id
+    )
+
+    if (connections.toString() !== articleVersion.connections.toString()) {
```
Comment (robertu7) [inline src/mutations/article/editArticle.ts:205]: same as above about stable array

File: src/mutations/article/editArticle.ts [hunk 3]
```diff
@@ -168,38 +171,45 @@ const resolver: GQLMutationResolvers['editArticle'] = async (
...
+      checkRevisionCount(article.revisionCount + 1)
+      updateRevisionCount = true
```
Comment (robertu7) [inline src/mutations/article/editArticle.ts:207]: we won't increase the revision count in previous implantation?

Comment (robertu7) [reply src/mutations/article/editArticle.ts:207]: and tags

**Review by robertu7 — APPROVED** [review]

File: src/mutations/article/editArticle.ts (hunk 3 above)
Comment (gary02) [reply src/mutations/article/editArticle.ts:207]: This PR only bugfix the increase revision count ingoring old data problem.

File: src/mutations/article/editArticle.ts (hunk 1 above)
Comment (gary02) [reply src/mutations/article/editArticle.ts:164]: I think tags ordering may matters ?

**Pull request merged**